src/main/kotlin/com/ampnet/identityservice/service/impl/AbstractBlockchainQueue.kt

```diff
@@ -41,48 +35,46 @@ class BlockchainQueueServiceImpl(
     }
 
     override fun destroy() {
-        logger.info { "Shutting down blockchain queue executor service..." }
+        logger.info { "Shutting down $queueName executor service..." }
         executorService.shutdown()
     }
 
-    override fun createWhitelistAddressTask(address: String, request: WhitelistRequest) {
-        if (isWhitelistedOrInProcess(address, request)) {
-            logger.info { "Address: $address for request: $request already whitelisted or in process" }
-            return
-        }
-        logger.info { "Received task for address: $address" }
-        val blockchainTask = BlockchainTask(address, request.issuerAddress, request.chainId, uuidProvider, timeProvider)
-        blockchainTaskRepository.save(blockchainTask)
-        logger.info { "Created BlockchainTask: $blockchainTask" }
-    }
+    protected abstract fun createBlockchainTaskFromPendingTask()
 
     @Suppress("TooGenericExceptionCaught")
     private fun processTasks() {
+        // creates pending tasks for all chains which have at least one address in the queue
+        createBlockchainTaskFromPendingTask()
+
         blockchainTaskRepository.getInProcess()?.let {
             try {
                 handleInProcessTask(it)
             } catch (ex: Throwable) {
-                logger.error("Failed to handle in process task: ${ex.message}")
-                blockchainTaskRepository.setStatus(it.uuid, BlockchainTaskStatus.FAILED, it.hash)
+                logger.error("[$queueName]Failed to handle in process task: ${ex.message}")
```

Comment:
You don't need to add `[$queueName]` here since it will be contained in the thread name of the used executor service.